src/class-wpml-gutenberg-integration.php

```diff
@@ -257,17 +257,30 @@ private function guess_inner_HTML_parts( $block ) {
 
 		$parts = array( $inner_HTML, '' );
 
-		preg_match( '#>\s*</#', $inner_HTML, $matches );
-
-		if ( count( $matches ) === 1 ) {
-			$parts = explode( $matches[0], $inner_HTML );
-			if ( count( $parts ) === 2 ) {
-				$match_mid_point = 1 + ( strlen( $matches[0] ) - 3 ) / 2;
-				// This is the first ">" char plus half the remaining between the tags
-
-				$parts[0]        .= substr( $matches[0], 0, $match_mid_point );
-				$parts[1]        = substr( $matches[0], $match_mid_point ) . $parts[1];
-			}
+		switch( $block->blockName ) {
+			case 'core/media-text':
+				$html_to_find = '<div class="wp-block-media-text__content">';
+				$pos          = strpos( $inner_HTML, $html_to_find ) + strlen( $html_to_find );
```

Comment:
@adrianosferreira, you should use multi-bytes functions `mb_strpos`, `mb_strlen` and `mb_substr`.

We also need to fix this in the `default` logic.

You can add some exotic chars in the tests.

Reply:
@strategio I don't see the need for that. I can see non-multibyte functions all over the place in the Gutenberg library and 0 occurrences of multibyte ones, so I guess we should change the whole lib. I guess that @BruceOnTheGo should have a reason for that.

Anyway, this code is just a suggestion from Bruce, I believe he had a reason to write using normal string functions.

Reply:
@adrianosferreira, I don't see disadvantage using multi-byte functions (some will complain about performance but it's not significant IMO), it's just safer and we already had a lot of bugs related to this.

If changing the functions to multi-byte in the code base does not break the existing tests, I would go for it.

Reply:
@strategio I keep forgetting to use the multi-byte functions. @adrianosferreira Please change to use the multi-byte functions. If the existing tests don't break we should be good to go.

Reply:
@strategio adjusted both suggestions, please check.